compiler/errors.go

```diff
@@ -16,3 +20,31 @@ func (c *Compiler) makeError(pos token.Pos, msg string) types.Error {
 func (c *Compiler) addError(pos token.Pos, msg string) {
 	c.diagnostics = append(c.diagnostics, c.makeError(pos, msg))
 }
+
+// errorAt returns an error value at the location of the instruction.
+// The location information may not be complete as it depends on debug
+// information in the IR.
+func errorAt(inst llvm.Value, msg string) scanner.Error {
```

Comment:
This is literally one of the built-in error type that is used in the go/* packages.

Yes I agree the error should be an `error` instead of a `string`, but that's not really compatible with the rest of the compiler at the moment. Eventually I'd like to streamline it across the compiler but that will take a while - to me, refactoring so that packages can be compiled independently is still higher priority.